### PR TITLE
tmc-langs as submodule and updated README

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -13,3 +13,6 @@
 [submodule "ext/tmc-checkstyle-runner"]
 	path = ext/tmc-checkstyle-runner
 	url = git@github.com:testmycode/tmc-checkstyle-runner.git
+[submodule "ext/tmc-langs"]
+	path = ext/tmc-langs
+	url = git@github.com:tmc-langs/tmc-langs.git
```
Comment (jamox) [inline .gitmodules]: Should probably point to testmycode, not to tmc-langs.

File: config/site.defaults.yml [hunk 2]
```diff
@@ -82,6 +82,8 @@ emails:
 # Allow people to create user accounts
 enable_signup: true
 
+# Enable tmc-langs framework
+enable_tmc_langs: false
```
Comment (jamox) [inline config/site.defaults.yml]: Setting for tmc-langs twice?

File: lib/tmc_langs.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'shellwords'
+require 'maven_project'
+
+class TmcLangs < MavenProject
+	def self.get
+		@instance ||= TmcLangs.new
```
Comment (jamox) [inline lib/tmc_langs.rb]: Indentation: 2 spaces

**Comment by jamox** [comment]
I would keep installation instructions generic for Readme, and then maybe have separate installation guide for ubuntu...

**Comment by PunyW** [comment]
Created new pull request with installation instructions in their own file

File: lib/tasks/submodules.rake [hunk 4]
```diff
@@ -2,10 +2,12 @@ require 'system_commands'
 require 'tmc_junit_runner'
 require 'tmc_checkstyle_runner'
 require 'tmc_comet'
+require 'tmc_langs'
```
Comment (mpartel) [inline lib/tasks/submodules.rake]: Does this exist?

File: lib/tasks/submodules.rake [hunk 5]
```diff
@@ -2,10 +2,12 @@ require 'system_commands'
...
 
 TmcJunitRunner.get.make_rake_tasks(self, 'junit_runner')
 TmcCheckstyleRunner.get.make_rake_tasks(self, 'checkstyle_runner')
 TmcComet.get.make_rake_tasks(self, 'comet')
+TmcLangs.get.make_rake_tasks(self, 'langs')
```
Comment (mpartel) [inline lib/tasks/submodules.rake]: Have you tested these?

Comment (mpartel) [inline lib/tasks/submodules.rake]: Never mind, sleepy...

**Comment by mpartel** [comment]
Please rebase away the back-and-forth commits.

**Comment by mpartel** [comment]
I get the following error:
```
$ git submodule update --init
Submodule 'ext/tmc-langs' (git@github.com:testmycode/tmc-langs.git) registered for path 'ext/tmc-langs'
Cloning into 'ext/tmc-langs'...
remote: Counting objects: 52, done.
remote: Total 52 (delta 0), reused 0 (delta 0)
Receiving objects: 100% (52/52), 8.76 KiB | 0 bytes/s, done.
Resolving deltas: 100% (12/12), done.
Checking connectivity... done.
fatal: reference is not a tree: 8a588feb44e7a66d27c36a7b5c000a29f9ad009a
Unable to checkout '8a588feb44e7a66d27c36a7b5c000a29f9ad009a' in submodule path 'ext/tmc-langs'
```

Does something need to be merged into testmycode/tmc-langs first?

**Comment by PunyW** [comment]
Rebased the back-and-forth commits, and fixed tmc-langs submodule referencing to wrong commit.

**Comment by jamox** [comment]
Not relevant atm

**Pull request closed without merging**